Sources/OktaIdx/Internal/Implementations/Version1/Responses/IDXClient+V1ResponseConstructors.swift

```diff
@@ -69,28 +69,33 @@ extension V1.Response {
         let authenticator: V1.Response.Authenticator
     }
 
-    func authenticatorState(for authenticatorId: String?) -> IDXClient.Authenticator.State {
-        guard let authenticatorId = authenticatorId else {
-            return .normal
-        }
-        
-        if currentAuthenticatorEnrollment?.value.id == authenticatorId {
-            return .enrolling
-        }
-        
-        else if currentAuthenticator?.value.id == authenticatorId {
-            return .authenticating
-        }
-        
-        else if authenticatorEnrollments?.value.first(where: { (authenticator) -> Bool in
-            authenticator.id == authenticatorId
-        }) != nil {
-            return .enrolled
-        }
+    func authenticatorState(for authenticators: [Authenticator],
+                            in jsonPaths: [String]) -> IDXClient.Authenticator.State
+    {
+        var state = [IDXClient.Authenticator.State]()
+        state = jsonPaths.reduce(into: state, { state, jsonPath in
+            switch jsonPath {
+            case "$.currentAuthenticatorEnrollment":
+                state.append(.enrolling)
+                
+            case "$.currentAuthenticator":
+                state.append(.authenticating)
+                
+            case "$.recoveryAuthenticator":
+                state.append(.recovery)
```

Comment:
Is it expected that we would have multiple of these? Why not just return early when one is found?

Reply:
Their specificity increases with higher enum values. Ultimately I'd prefer this to be a bitmask, but I need to ensure Objective-C compatibility. This is another tech debt item on my backlog